torch/_inductor/lowering.py

```diff
@@ -2926,6 +2926,11 @@ def _unsafe_index_put_(self, indices, values, accumulate=False):
     return index_put_impl_(self, indices, values, accumulate, check=False)
 
 
+def needs_fallback_due_to_atomic_add_limitations(dtype):
+    # tl.atomic_add does NOT support the following types
+    return dtype in {torch.int64, torch.bool, torch.bfloat16}
```

Comment:
Shouldn't this check the device as well?

Reply:
Yeah, i guess we should only do the pessimization in cuda mode and not cpu mode.